### PR TITLE
Port almost all integration tests to use Kernel API

### PR DESCRIPTION
**Pull request by pontusmelke**
Update all of the integration tests that does not use `nodeUniqueIndexSeek` to use Kernel API.

Builds on and replaces https://github.com/neo4j/neo4j/pull/11243/commits/d4611966e5e6e2eaa5efe94da5b3b9b83a06aec4

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java [hunk 1]
```diff
@@ -107,12 +109,13 @@
...
     private final Lazy<ExplicitIndexTransactionState> explicitIndexes;
     private final Procedures procedures;
+    private final SchemaState schemaState;
 
     public AllStoreHolder( StorageEngine engine,
             StorageStatement statement,
             KernelTransactionImplementation ktx,
             DefaultCursors cursors,
-            ExplicitIndexStore explicitIndexStore, Procedures procedures )
+            ExplicitIndexStore explicitIndexStore, Procedures procedures  )
```
Comment (sherfert) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java]: Oh no, an extra space!

File: enterprise/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/PropertyConstraintValidationIT.java [hunk 2]
```diff
@@ -412,21 +412,21 @@ public void shouldAllowCreationOfNonConflictingData() throws Exception
...
+            createEntity( transaction, "Type1", "key1", "value3" );
 
             commit();
 
             // then
             assertEquals( 5, entityCount() );
         }
 
-        protected int countNodes( KernelTransaction transaction )
+        int countNodes( KernelTransaction transaction )
```
Comment (sherfert) [inline enterprise/kernel/src/test/java/org/neo4j/kernel/impl/api/integrationtest/PropertyConstraintValidationIT.java]: Can we share this code with `KernelIntegrationTest` ?

**Review by sherfert — APPROVED** [review]
Looks good.

Is the old kernel API still used somewhere in production code? If so, I am a little concerned removing all this test coverage from the old API...

Please have a look at the outdated comment, I think it's actually not outdated.

File: community/kernel/src/main/java/org/neo4j/storageengine/api/StorageEngine.java [hunk 3]
```diff
@@ -41,6 +42,12 @@
      */
     StoreReadLayer storeReadLayer();
 
+    /**
+     * Return access to the schema state
+     * @return an interface for accessing schema state
+     */
+    SchemaState schemaState();
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/storageengine/api/StorageEngine.java]: it should not be exposed here. As you can see already in the bottom of this interface we have already some methods that are marked to be removed - one of them is schema cache loader and new method just extend amount of things that should not be here.

File: community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java [hunk 4]
```diff
@@ -125,6 +130,7 @@ public AllStoreHolder( StorageEngine engine,
         this.properties = statement.properties();
         this.explicitIndexStore = explicitIndexStore;
         this.procedures = procedures;
+        this.schemaState = engine.schemaState();
```
Comment (MishaDemianenko) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java]: as already mentioned storage engine should not be used as schema state provider.

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]

**Comment by pontusmelke** [comment]
@MishaDemianenko @sherfert you should be happier now

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**